### ISSUE TITLE
Inconsistent window crashing bug

### ISSUE THREAD

**Issue by MrVallentin** [glfw/glfw#803]
I've found an inconsistent bug, that results in an instant crash. The crash happens when I press any media buttons on my keyboard.

I have a [Corsair K70 RGB keyboard](http://i.imgur.com/9Tblcsc.jpg). It's the media keys in the top right corner, that causes the crashing.

It should be mentioned that I'm not trying to detect whether any of the media keys are pressed, I was just casually pausing my background music, while having my application running and it crashed.

I wanted to figure out what caused it, so I naturally tried getting it to crash again.
I would launch my application press the <kbd>Play/Pause</kbd> media key and it would crash. The weird and inconsistent part is that sometimes it happened 5 times in a row, of launching, press and crash. But sometimes it didn't crash.

If it didn't crash by the first press, I tried moving the window and pressing the button, which also caused it to crash a few times. All in all, after continuously pressing the <kbd>Play/Pause</kbd> key it would eventually crash the application (something like 9/10 times).

**I don't consider this a crucial bug, but it is a rather peculiar one.**

I do have a GLFW error callback, but it didn't print anything before crashing.

```c
static void errorCallback(int error, const char *description)
{
	printf("GLFW Error: [%d]: %s\n", error, description);
}
```

I'm using Visual Studio Express 2013, and at the time of the crash the debugger would either show/go here mostly:

[![](http://i.imgur.com/74D91MY.png)](http://i.imgur.com/74D91MY.png)

and a few times:

[![](http://i.imgur.com/ebJIcVk.png)](http://i.imgur.com/ebJIcVk.png)

I'm using GLFW 3.2.0.

**Comment by MrVallentin** [comment]
This issue can be closed, as I figured out the problem and it's actually entirely my fault.

For some reason I never checked whether the key returned by the key callback was `GLFW_KEY_UNKNOWN`.

Further I have a wrapper, to simplify GLFW, which looks like this:

```cpp
class GLWindow
{
    GLFWwindow *window;

    bool keyDownBegin[GLFW_KEY_LAST + 1];

    ...
```

Which is (obviously) why the debugger would trigger, as modifying `keyDownBegin[-1]` would modify parts of `window` (considering no padding was added).
